src/package/package.ts

```diff
@@ -41,131 +46,206 @@ const messages = Messages.loadMessages('@salesforce/packaging', 'package');
  * Package class.
  *
  * This class provides the base implementation for a package.
+ * To create a new instance of a package, use the static async Package.create({connection, project, packageOrAliasId}) method.
  */
-export class Package extends AsyncCreatable<PackageOptions> implements IPackage {
+export class Package {
+  private packageId: string;
+  private packageData: PackagingSObjects.Package2;
   public constructor(private options: PackageOptions) {
-    super(options);
+    this.init();
   }
 
   /**
-   * Given a Salesforce ID for a package resource and the type of resource,
-   * ensures the ID is valid.
+   * Create a new instance of a package, using supplied options.
    *
-   * Valid ID types and prefixes for packaging resources:
-   * 1. package ID (0Ho)
-   * 2. subscriber package version ID (04t)
-   * 3. package install request ID (0Hf)
-   * 4. package uninstall request ID (06y)
-   *
-   * @param id Salesforce ID for a specific package resource
-   * @param type The type of package ID
+   * @param options
    */
-  public static validateId(id: string, type: PackageIdType): void {
-    const prefix = packagePrefixes[type];
-    if (!id.startsWith(prefix)) {
-      throw messages.createError('invalidPackageId', [type, id, prefix]);
-    }
-    if (!sfdc.validateSalesforceId(id)) {
-      throw messages.createError('invalidIdLength', [type, id]);
-    }
+  public static async create(options: PackageOptions): Promise<Package> {
+    return new Package(options);
```

Comment:
If we want to be consistent with `PackageVersion` then `createPackage` code should move here and that function goes away.